Added initial generators qunit and newqunittest

georgbischoff opened the pull request.

ArnaudBuchholz commented on generators/newqunittest/templates/test/unit/$testFile.js:
```diff
@@ -0,0 +1,32 @@
+			//assert.equal(<actual result>, <expected result>, <optional message>);
+			//assert.ok(<actual result>, <optional message>);
+			assert.ok(false, "Implement me");
+		});<% } %>		
```
Would that makes sense to also insert an example of asynchronous test ?

georgbischoff replied:
Great idea. I added two options for the testing of asynchronous operations and errorThrows

ArnaudBuchholz commented on generators/qunit/templates/test/testsuite.qunit.js:
```diff
@@ -0,0 +1,11 @@
+        sContextPath = location.pathname.substring(0, location.pathname.lastIndexOf("/") + 1);
+        
+    oSuite.addTestPage(sContextPath + "unit/unitTests.qunit.html");
+    <% if (typeof(opa5Journeys) !== "undefined") { %>oSuite.addTestPage(sContextPath + "integration/opaTests.qunit.html");
```
Would that make sense to apply the journey splitting pattern ?

georgbischoff replied:
We can discuss that for the opa-generator.

ArnaudBuchholz commented on test/qunit.js:
```diff
@@ -0,0 +1,64 @@
+const assert = require("yeoman-assert");
```
Nice that you can test the generator

ArnaudBuchholz reviewed (commented):
I am not very familiar with yeoman generators and it is hard to read through the proposed options.
I just noted two possible improvements.

IObert commented:
Thanks for your PR and the work you put in it. I just tested the generator and noticed the following things:

* I wanted to run project:newqunittest first as I assumed that all files needed for qunit tests are already present. And I was able to run the subgenerator without any errors but I noticed that it didn't create useful files:
<img width="1371" alt="Screenshot 2021-08-18 at 15 29 03" src="https://user-images.githubusercontent.com/1873245/129910401-c8dcb1be-04ac-4d59-bb6a-da34ee17114b.png">
 => I think it'd be nice if the subgenerator wouldn't fail if the needed files to run it aren't there


* Then, I ran  project:qunit in a new project which added a few more files. Once the files were generated, I ran `npm test` and noticed that the new files break the following es-lint rules:
<img width="1226" alt="Screenshot 2021-08-18 at 15 46 56" src="https://user-images.githubusercontent.com/1873245/129910718-549a669c-e347-4356-a74c-3f51fc114f1d.png">

=> Can you please adjust the files so that the test completes succcessfully

georgbischoff commented:
- Fixed the es-lint issues
- Added selection uimodules to generate single qunit test in the right folder. Also added a remark to run the qunit generator if the overall test-suite was not yet generated.

IObert commented:
Thanks a lot!
